### PR TITLE
3603 move test suite api v2 viz

### PR DESCRIPTION
**Pull request by juanignaciosl**
@Kartones CR this fix for #3603, please.

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3685/))

File: spec/requests/api/json/visualizations_controller_shared_examples.rb [hunk 1]
```diff
@@ -1078,6 +1078,47 @@ def api_visualization_creation(user, headers, additional_fields = {})
...
+        last_response.status.should == 200
+
+        visualization = JSON.parse(last_response.body)
+
+        # TODO: this endpoint doesn't exist now. Current replacement?
+        #get "/api/v1/viz/#{visualization.fetch('id')}/viz?api_key=#{@api_key}",
+        #  {}, @headers
+        #last_response.status.should == 403
+
+        get "/api/v2/viz/#{visualization.fetch('id')}/viz?api_key=#{@api_key}",
```
Comment (Kartones) [inline spec/requests/api/json/visualizations_controller_shared_examples.rb]: mmm we should remove handcrafted calls and use instead xxx_url rails helpers, for better routing and support of subdomainless urls

**Comment by Kartones** [comment]
Definetly I'd rewrite them, we need to wipe out all handcrafted urls...

**Comment by juanignaciosl** [comment]
Ok, I'll rewrite the whole spec.

**Comment by juanignaciosl** [comment]
@Kartones take a look at the handcrafted url removal commit :)

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3689/))

**Comment by Kartones** [comment]
:+1:

**Pull request merged**